lib/bgs/exceptions/bgs_errors.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'bgs/exceptions/service_exception'
+module BGS
+  module Exceptions
+    module BGSErrors
+      include SentryLogging
+      MAX_ATTEMPTS = 3
+
+      def with_multiple_attempts_enabled
+        attempt ||= 0
+        yield
+      rescue => e
+        attempt += 1
+        if attempt < MAX_ATTEMPTS
+          notify_of_service_exception(e, __method__.to_s, attempt, :warn)
+          retry
+        end
+
+        notify_of_service_exception(e, __method__.to_s)
+      end
+
+      def raise_backend_exception(key, source, error)
+        exception = BGS::ServiceException.new(
+          key,
+          { source: source.to_s },
+          403,
```

Comment:
Are all the exceptions 403?

Reply:
I combed the documentation and there is no exception documentation. My hope was to get the error handling more sophisticated as we encountered them.